pkg/gui/controllers/helpers/refs_helper.go

```diff
@@ -119,31 +119,38 @@ func (self *RefsHelper) ResetToRef(ref string, strength string, envVars []string
 	return nil
 }
 
-func (self *RefsHelper) CreateSortOrderMenu(onSelected func(sortOrder string) error) error {
-	type sortOrderWithKey struct {
-		key       types.Key
-		label     string
-		sortKey   string
-		sortOrder string
+func (self *RefsHelper) CreateSortOrderMenu(sortOptionsOrder []string, onSelected func(sortOrder string) error) error {
+	type sortMenuOption struct {
+		key         types.Key
+		label       string
+		description string
+		sortOrder   string
 	}
-	sortKeys := []sortOrderWithKey{
-		{label: self.c.Tr.SortAlphabetical, sortKey: "refname", sortOrder: "alphabetical", key: 'a'},
-		{label: self.c.Tr.SortByDate, sortKey: "-committerdate", sortOrder: "date", key: 'd'},
+	availableSortOptions := map[string]sortMenuOption{
+		"recency":      {label: self.c.Tr.SortByRecency, description: self.c.Tr.SortBasedOnReflog, key: 'r'},
+		"alphabetical": {label: self.c.Tr.SortAlphabetical, description: "--sort=refname", key: 'a'},
+		"date":         {label: self.c.Tr.SortByDate, description: "--sort=-committerdate", key: 'd'},
+	}
+	sortOptions := make([]sortMenuOption, 0, len(sortOptionsOrder))
+	for _, key := range sortOptionsOrder {
+		if sortOption, ok := availableSortOptions[key]; ok {
+			sortOption.sortOrder = key
+			sortOptions = append(sortOptions, sortOption)
+		}
```

Comment:
Since it's a programming error to pass a string that doesn't occur in `availableSortOptions`, I'd add an `else` with a panic here (which seems to be go's closest equivalent to an assert in other languages).

Reply:
Fixed in https://github.com/jesseduffield/lazygit/commit/bea96a5babac6ae2a941e99a01446d7c23fc2c8e#diff-5ce8d32b6a73dd700e90181f144d5142ec388b1142ae13db669773b8c928dd95R137